src/lib/services/evm/jsonRpc.ts

```
@@ -34,3 +35,42 @@ export const requestJsonRpc = (
 
       return parsed;
     });
+
+export const requestBatchJsonRpc = (
+  endpoint: string,
+  requests: {
+    method: string;
+    params: JsonRpcParams[];
+  }[]
+) =>
+  axios
+    .post(
+      endpoint,
+      requests.map(({ method, params }, index) => ({
+        jsonrpc: "2.0",
+        id: index,
+        method,
+        params,
+      })),
+      {
+        headers: {
+          "Content-Type": "application/json",
+        },
+      }
+    )
+    .then(({ data }) => {
+      const parsed = parseWithError(zBatchJsonRpcResponse, data);
+      if (!Array.isArray(parsed)) {
+        throw new Error(parsed.error.message);
+      }
+
+      const parsedResults = new Array<JsonRpcResponse["result"]>(
+        requests.length
+      );
+      parsed.forEach((result) => {
+        if (typeof result.id === "number")
+          parsedResults[result.id] = result.result;
```

If result.id is not number, let's throw an error.